topiary-config/src/serde.rs

```diff
@@ -0,0 +1,229 @@
+    /// "\t", etc.)
+    pub indent: Option<String>,
+}
+
+// TODO I don't think we're going to need this here...but maybe
+impl Language {
+    #[cfg(not(wasm))]
+    pub fn find_query_file(&self) -> TopiaryConfigResult<PathBuf> {
+        let basename = PathBuf::from(match self.name.as_str() {
+            "bash" => "bash",
+            "json" => "json",
+            "nickel" => "nickel",
+            "ocaml" | "ocaml_interface" => "ocaml",
+            "ocamllex" => "ocamllex",
+            "rust" => "rust",
+            "toml" => "toml",
+            "tree_sitter_query" => "tree-sitter-query",
+            name => return Err(TopiaryConfigError::UnknownLanguage(name.to_string())),
+        })
+        .with_extension("scm");
+
+        #[rustfmt::skip]
+        let potentials: [Option<PathBuf>; 4] = [
+            std::env::var("TOPIARY_LANGUAGE_DIR").map(PathBuf::from).ok(),
+            option_env!("TOPIARY_LANGUAGE_DIR").map(PathBuf::from),
+            Some(PathBuf::from("./queries")),
+            Some(PathBuf::from("../queries")),
+        ];
+
+        potentials
+            .into_iter()
+            .flatten()
+            .map(|path| path.join(&basename))
+            .find(|path| path.exists())
+            .ok_or_else(|| TopiaryConfigError::QueryFileNotFound(basename))
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn grammar(&self) -> TopiaryConfigResult<topiary_tree_sitter_facade::Language> {
+        Ok(match self.name.as_str() {
+            "bash" => tree_sitter_bash::language(),
+            "json" => tree_sitter_json::language(),
+            "nickel" => tree_sitter_nickel::language(),
+            "ocaml" => tree_sitter_ocaml::language_ocaml(),
+            "ocaml_interface" => tree_sitter_ocaml::language_ocaml_interface(),
+            "ocamllex" => tree_sitter_ocamllex::language(),
+            "rust" => tree_sitter_rust::language(),
+            "toml" => tree_sitter_toml::language(),
+            "tree_sitter_query" => tree_sitter_query::language(),
+            name => return Err(TopiaryConfigError::UnknownLanguage(name.to_string())),
+        }
+        .into())
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    pub async fn grammar(&self) -> TopiaryConfigResult<topiary_tree_sitter_facade::Language> {
+        let language_name = match self.name.as_str() {
+            "bash" => "bash",
+            "json" => "json",
+            "nickel" => "nickel",
+            "ocaml" => "ocaml",
+            "ocaml_interface" => "ocaml_interface",
+            "ocamllex" => "ocamllex",
+            "rust" => "rust",
+            "toml" => "toml",
+            "tree_sitter_query" => "query",
+            name => return Err(TopiaryConfigError::UnknownLanguage(name.to_string())),
+        };
+
+        Ok(topiary_web_tree_sitter_sys::Language::load_path(&format!(
+            "/playground/scripts/tree-sitter-{language_name}.wasm"
```

Comment:
I guess hardcoding the path is fine, as it matches the layout of the website/playground, which is still under control of this repo.